access_log/access_log_record.go

```diff
@@ -94,3 +100,15 @@ func (r *AccessLogRecord) LogMessage() string {
 	recordBuffer := r.makeRecord()
 	return recordBuffer.String()
 }
+
+func (r *AccessLogRecord) ExtraHeaders() string {
+	extraHeadersToLog := make(map[string]string)
+	for _, header := range r.ExtraHeadersToLog {
+		extraHeadersToLog[header] = r.FormatRequestHeader(header)
+	}
+	marshalledExtraHeadersToLog, err := json.Marshal(extraHeadersToLog)
+	if err != nil {
+		return "{}"
```

Comment:
I think we might want to return something to the effect of "UnableToMarshalJson" instead of an empty hash here, in keeping with some of the other fields. That way a developer/operator would no something is wrong with their logging instead of being confused as to why the extra headers did not show up.